### ISSUE TITLE
Save decoded GRPC/Protocol Buffer content with mitmdump

### ISSUE THREAD

**Issue by FreelancerCGN** [mitmproxy/mitmproxy#6870]
#### Problem Description
I would like to save the decoded GRPC/Protocol Buffer content with mitmdump.

I called mitmdump with

`/usr/local/bin/mitmdump --listen-host <IP ADDRESS> --listen-port <PORT> --mode upstream:http://<IP>:<PORT> --no-rawtcp --scripts save.py --set block_global=false --set http3=false --set flow_detail=3 --set dumper_default_contentview=hex`

At the script save.py I tried almost everything I found here and at StackOverflow etc. I managed only to save the encoded binary content, not the decoded protobuf result (similar to the dump output of mitmdump).

Examples (not working):

```
import mitmproxy
import time
from mitmproxy import http

# def response(flow: http.HTTPFlow) -> None:
def response(flow):

if flow.request.pretty_url.endswith("test"):

   timestr = time.strftime("%Y%m%d-%H%M%S")
   with open("log/" + timestr + "-mitm-response-test.log", "wb") as f:
  
      # f.write(flow.response.content) # doesn't work
      # f.write(flow.response.text) # doesn't work
      # f.write(flow.response.get_content(False)) # doesn't work
      # f.write(flow.response.get_text()) # doesn't work
      # f.write(flow.response.get_text(False).encode()) # doesn't work
```

It's clearly a HTTP/2.0 request and a HTTP/2.0 response, no HTTP/1.1. And it's a GRPC/Protocol Buffer request/response, no "classic" Protocol Buffer.

How to solve this problem?

#### Steps to reproduce the behavior:
1. 
2. 
3. 

#### System Information
Mitmproxy: 10.3.0
Python:    3.10.12
OpenSSL:   OpenSSL 3.2.1 30 Jan 2024
Platform:  Linux-5.15.0-107-generic-x86_64-with-glibc2.35

**Comment by mhils** [comment]
I don't think we have a good way to export the pretty-printed result from a contentview at the moment. But this does definitely sound useful!
